src/AppInstallerRepositoryCore/Microsoft/Schema/1_3/Interface_1_3.cpp

```diff
@@ -0,0 +1,92 @@
+    {
+        return { 1, 3 };
+    }
+
+    void Interface::CreateTables(SQLite::Connection& connection)
+    {
+        SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, "createtables_v1_3");
+
+        V1_2::Interface::CreateTables(connection);
+
+        V1_0::ManifestTable::AddColumn(connection, { HashVirtualTable::ValueName(), HashVirtualTable::SQLiteType() });
+
+        savepoint.Commit();
+    }
+
+    SQLite::rowid_t Interface::AddManifest(SQLite::Connection& connection, const Manifest::Manifest& manifest, const std::filesystem::path& relativePath)
+    {
+        SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, "addmanifest_v1_3");
+
+        SQLite::rowid_t manifestId = V1_2::Interface::AddManifest(connection, manifest, relativePath);
+
+        // Set the hash value if provided
+        if (!manifest.StreamSHA256.empty())
+        {
+            THROW_HR_IF(E_INVALIDARG, manifest.StreamSHA256.size() != Utility::SHA256::HashBufferSizeInBytes);
+            V1_0::ManifestTable::UpdateValueIdById<HashVirtualTable>(connection, manifestId, manifest.StreamSHA256);
+        }
+
+        savepoint.Commit();
+
+        return manifestId;
+    }
+
+    std::pair<bool, SQLite::rowid_t> Interface::UpdateManifest(SQLite::Connection& connection, const Manifest::Manifest& manifest, const std::filesystem::path& relativePath)
+    {
+        SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, "updatemanifest_v1_3");
+
+        auto [indexModified, manifestId] = V1_2::Interface::UpdateManifest(connection, manifest, relativePath);
+
+        // Set the hash value if provided
+        if (!manifest.StreamSHA256.empty())
+        {
+            THROW_HR_IF(E_INVALIDARG, manifest.StreamSHA256.size() != Utility::SHA256::HashBufferSizeInBytes);
+
+            auto currentHash = std::get<0>(V1_0::ManifestTable::GetIdsById<HashVirtualTable>(connection, manifestId));
+
+            if (currentHash.size() != Utility::SHA256::HashBufferSizeInBytes ||
+                !std::equal(currentHash.begin(), currentHash.end(), manifest.StreamSHA256.begin()))
+            {
+                V1_0::ManifestTable::UpdateValueIdById<HashVirtualTable>(connection, manifestId, manifest.StreamSHA256);
+                indexModified = true;
+            }
+        }
+
+        savepoint.Commit();
+
+        return { indexModified, manifestId };
+    }
+
+    std::optional<std::string> Interface::GetPropertyByManifestIdInternal(const SQLite::Connection& connection, SQLite::rowid_t manifestId, PackageVersionProperty property) const
+    {
+        switch (property)
+        {
+        case AppInstaller::Repository::PackageVersionProperty::ManifestSHA256Hash:
+        {
+            SQLite::blob_t hash = std::get<0>(V1_0::ManifestTable::GetIdsById<HashVirtualTable>(connection, manifestId));
+            return hash.empty() ? std::optional<std::string>{} : Utility::SHA256::ConvertToString(hash);

```

Comment:
>std::optional<std::string>{} [](http://example.com/codeflow?start=34&length=28)

{}

Reply:
I didn't actually try that, but I suspect that would create an empty string to match the return type of the other half of the ternary.  Then instead of an empty optional, we would return a filled optional with an empty string, which is not what we want.